compiler-core/src/javascript/tests/snapshots/gleam_core__javascript__tests__case__case_on_error.snap

```diff
@@ -0,0 +1,30 @@
+---
+source: compiler-core/src/javascript/tests/case.rs
+expression: "\nfn a_result() { Error(1) }\n\npub fn main() {\n  case a_result() {\n    Error(_) -> 1\n    _ -> 2\n  }\n}"
+---
+----- SOURCE CODE
+
+fn a_result() { Error(1) }
+
+pub fn main() {
+  case a_result() {
+    Error(_) -> 1
+    _ -> 2
+  }
+}
+
+----- COMPILED JAVASCRIPT
+import { Error } from "../gleam.mjs";
+
+function a_result() {
+  return new Error(1);
+}
+
+export function main() {
+  let $ = a_result();
+  if (!$.isOk()) {
```

Comment:
I think we can remove the special case for result and use instanceof like the other variants?

Reply:
We would still need to special case results as we'd have to make sure that we import the constructors from the prelude to perform the `instanceOf` check. But if you want I can switch to instance of checks for these as well

Reply:
Yes please